onnxscript/function_libs/torch_aten/graph_building.py

```diff
@@ -0,0 +1,446 @@
+    # TODO(justinchuby): Do we really need this?
+    if isinstance(tensors, Sequence):
+        return [_unwrap_tensor_to_torch_value(output) for output in tensors]
+    return _unwrap_tensor_to_torch_value(tensors)
+
+
+class TorchScriptEvaluator(evaluator.Evaluator):
+    """An onnxscript Evaluator that captures the graph into torchscript."""
+
+    def __init__(self, graph: TorchScriptGraph):
+        self._graph: TorchScriptGraph = graph
+
+    @property
+    def graph(self) -> TorchScriptGraph:
+        return self._graph
+
+    @beartype
+    def eval_function(
+        self,
+        function: onnxscript.OnnxFunction,
+        args: Sequence[ValidArgumentType],
+        kwargs: Sequence[ValidArgumentType],
+    ):
+        # args/kwargs are TorchScriptTensor/python built-in based
+        inputs, attributes = _split_args_kwargs_to_input_attr(function, args, kwargs)
+        return self._graph.add_function(function, inputs, attributes)
+
+    def _eval(self, schema: onnx.defs.OpSchema, inputs, attributes, closure: Any):
+        # TODO(justinchuby): Does it really know what the inputs are?
+
+        del closure  # Unused
+        return self._graph.add_op(schema, inputs, attributes)
+
+
+@beartype
+def _add_attribute_to_torchscrpt_node(
+    node: torch.Node, key: str, value: float | int | str | Sequence[float] | Sequence[int]
+):
+    """Initializes the right attribute based on type of value."""
+    if isinstance(value, float):
+        return node.f_(key, value)
+    if isinstance(value, int):
+        return node.i_(key, value)
+    if isinstance(value, str):
+        return node.s_(key, value)
+    if isinstance(value, torch.Tensor):
+        return node.t_(key, value)
+    if isinstance(value, Sequence):
+        if isinstance(value, float):
+            return node.fs_(key, list(value))
+        if isinstance(value, int):
+            # Need to use getattr because 'is' is a reserved keyword
+            return getattr(node, "is_")(key, list(value))
+        raise TypeError(f"Unsupported sequence type '{type(value)}' for attribute '{key}'")
+    raise TypeError(f"Unsupported attribute type '{type(value)}' for attribute '{key}'")
+
+
+@beartype
+def _create_torchscript_op(
+    graph: torch.Graph,
+    opname: str,
+    *,
+    inputs: Sequence[torch.Value],
+    attributes: dict[str, Any],
+    n_outputs: int = 1,
+) -> tuple[torch.Value, ...]:
+    # Filter out None attributes, this can be convenient client side because
+    # now they can pass through None attributes, and have them not show up
+    attributes = {k: v for k, v in attributes.items() if v is not None}
+
+    node = graph.create(opname, inputs, n_outputs)
+    node = graph.insertNode(node)
+    node_ouputs = tuple(node.outputs())
+
+    assert len(node_ouputs) == n_outputs
+    # Add all attributes
+    for key, value in sorted(attributes.items()):
+        _add_attribute_to_torchscrpt_node(node, key, value)
+
+    return node_ouputs
+
+
+class TorchScriptGraph:
```

Comment:
Please rename `TorchScriptGraph` to `TorchScriptWrapper`. I thought it `torch._C.Graph` until seeing this definition.

Reply:
This is an onnxscript graph. We will consider a better name